packages/block-editor/src/components/spacing-sizes-control/style.scss

```diff
@@ -0,0 +1,121 @@
+.component-spacing-sizes-control {
+	box-sizing: border-box;
+	padding-bottom: 12px;
+	width: 100%;
+	display: contents;
+
+	legend {
+		padding-left: 0;
+		grid-column: 1 / 1;
+		grid-row: 1 / 1;
+	}
+
+	.components-spacing-sizes-control__side-labels {
+		grid-column: 1 / 1;
+		min-height: 30px;
+	}
+
+	.components-spacing-sizes-control__hint-single {
+		margin-top: 0;
+	}
+
+	.components-spacing-sizes-control__hint-single,
+	.components-spacing-sizes-control__hint-all {
+		color: $gray-700;
+		font-size: 12px;
+	}
+
+	.components-spacing-sizes-control__hint-all {
+		grid-column: 1 / 1;
+		justify-self: left;
+		grid-row: 1 / 1;
+		position: relative;
+		top: 1rem;
+		left: 0;
+	}
+
+	.components-spacing-sizes-control__custom-toggle-all {
+		grid-column: 2 / 2;
+		grid-row: 1 / 1;
+		justify-self: end;
+		margin-right: $grid-unit-05;
+	}
+
+	.components-spacing-sizes-control__custom-toggle-single {
+		grid-column: 3 / 3;
+		justify-self: end;
+	}
+
+	.component-spacing-sizes-control__linked-button {
+		grid-column: 3 / 3;
+		grid-row: 1 / 1;
+		justify-self: end;
+	}
+
+	.components-spacing-sizes-control__custom-value-range {
+		grid-column: span 2;
+		margin-left: $grid-unit-10;
+		padding-right: $grid-unit-10;
+		height: 30px;
+	}
+
+	.components-spacing-sizes-control__range-control {
+		grid-column: span 3;
+		padding-right: $grid-unit-10;
+		height: 30px;
+	}
+
+	.components-range-control__wrapper {
+		margin-bottom: 0;
+	}
+
+	.components-range-control__mark {
+		height: $grid-unit-05;
+		width: 3px;
+		background-color: #fff;
+		z-index: 1;
+	}
+
+	.components-range-control__marks {
+		margin-top: 17px;
+	}
+
+	[class*="ThumbWrapper-thumbColor"] {
+		z-index: 3;
+	}
```

Comment:
To be honest this level of internal restyling makes me a bit nervous 😅 There are some changes coming down the pipeline to the RangeControl component (#42315 #41973), so this is especially prone to break. If this were a third-party consumer, it would be their responsibility to prevent breakage (e.g. add visual regression tests, add Storybook stories for easier testing of different states). But in this repo, in reality, they either just break because nobody notices, or it shifts the maintenance cost "upstream" to wp-components to ensure that all these hacks don't break after a code change.

I'm wondering if there is any way we could mitigate this. Is the plan to merge this style variant into RangeControl eventually? (Apologies if you've already coordinated with @aaronrobertshaw on this!)

Reply:
Yep, makes me nervous too 😄 , but have been talking with @aaronrobertshaw about this and he is looking at how this concept of a segmented range control can be incorporated into his refactor of the range component - will closely co-ordinate with him on that to make sure we don't end up with a breakage when the new version is merged.

Reply:
Sounds great, thanks!